ophyd/areadetector/base.py

```diff
@@ -19,6 +20,77 @@ def __init__(self, prefix, **kwargs):
         super().__init__(prefix + '_RBV', write_pv=prefix, **kwargs)
 
 
+class NDDerivedSignal(DerivedSignal):
+    """
+    DerivedSignal to shape a flattened array
+
+    The purpose of this class is to take a flattened array and shape in its'
```

Comment:
Can you please write a numpydoc style docstring?  The `shape` input should be carefully explained.

Reply:
Will do.